### PR TITLE
Use env for config if possible

### PR DESCRIPTION
**Pull request by sthapa**
Updates to allow environment to configuration various settings

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -11,7 +11,7 @@ ServiceX REST Server
 This is a flask app for receiving user requests and orchestrating the ServiceX
 services:
 
-.. image:: doc/sequence_diagram.png
+.. image:: https://servicex.readthedocs.io/en/latest/img/ServiceX-request-lifecycle.png
```
Comment (AndrewEckart) [inline README.rst:14]: Please add a white background; transparency makes the diagram unreadable if you use GitHub in dark mode:
image

Consider linking directly to the image at https://github.com/ssl-hep/ServiceX/blob/develop/docs/img/ServiceX-request-lifecycle.png instead of indirectly via RtD.

Change doesn't seem related to the rest of the PR but that's not a big deal.

File: servicex/transformer_manager.py [hunk 2]
```diff
@@ -97,13 +98,19 @@ def create_job_object(request_id, image, chunk_size, rabbitmq_uri, workers,
             env = env + [env_var_instance_name]
 
         if result_destination == 'object-store':
+            minio_url = os.environ.get('MINIO_URL',
+                                       current_app.config['MINIO_URL_TRANSFORMER'])
+            access_key = os.environ.get('MINIO_ACCESS_KEY',
+                                        current_app.config['MINIO_ACCESS_KEY'])
+            secret_key = os.environ.get('MINIO_SECRET_KEY',
+                                        current_app.config['MINIO_SECRET_KEY'])
```
Comment (AndrewEckart) [inline servicex/transformer_manager.py]:
```suggestion
            cfg = current_app.config
            minio_url = os.environ.get('MINIO_URL', cfg['MINIO_URL_TRANSFORMER'])
            access_key = os.environ.get('MINIO_ACCESS_KEY', cfg['MINIO_ACCESS_KEY'])
            secret_key = os.environ.get('MINIO_SECRET_KEY', cfg['MINIO_SECRET_KEY'])
```

**Review by AndrewEckart — APPROVED** [review]
Looks good, thanks! I especially like that it follows the [convention](https://stackoverflow.com/questions/11077223/what-order-of-reading-configuration-values) that environment variables take precedence over configuration files.

**Comment by AndrewEckart** [comment]
Thought this could be a separate PR, I think we need to add some documentation of all the possible configuration values and which can be overridden using environment variables.

File: README.rst (hunk 1 above)
Comment (sthapa) [reply README.rst:14]: What's the advantages of a direct link?  I understand that it gets rid of a layer of indirection, but I think that maybe we want that in case we change where the RtD images come from?

File: servicex/transformer_manager.py (hunk 2 above)
Comment (sthapa) [reply servicex/transformer_manager.py]: Used a different method so this isn't relevant anymore.  I'm just using current_app.config['BLAH'] without querying the enivon.

File: README.rst (hunk 1 above)
Comment (AndrewEckart) [reply README.rst:14]: The RtD link relies on Read the Docs being up, the success of our latest build, the repo RtD is linked to, and the path to the image in that repo.

The direct link relies only on the repo and path, not RtD. If we were to point RtD at another repo (theoretically - not likely), that repo would need to have the same exact path as the current repo in order for this link to continue working. So an indirect link via RtD is effectively tightly coupled with the repo anyway, since the path ends up in the URL, but adds an extra dependency on RtD.

Comment (sthapa) [reply README.rst:14]: Okay, is the link you provided correct?  Going to https://github.com/ssl-hep/ServiceX/blob/develop/docs/img/ServiceX-request-lifecycle.png  in my browser or httpie gives an html page.  How do I get a raw link?

Comment (AndrewEckart) [reply README.rst:14]: https://raw.githubusercontent.com/ssl-hep/ServiceX/develop/docs/img/ServiceX-request-lifecycle.png

Comment (BenGalewsky) [reply README.rst:14]: @sthapa - yes, please move this to a separate PR - it doesn't have anything to do with picking up config from environment vars, right?

Comment (sthapa) [reply README.rst:14]: No, it's a separate issue.  I'll move this to another PR

**Comment by BenGalewsky** [comment]
Looks like ` servicex_app.egg-info` directory got included in this PR as well.
@sthapa - I'm going move the commit we need to another PR and start over this PR has become very unruly!
